pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/impl/antlr4/AntlrNameDictionary.java

```diff
@@ -94,77 +96,61 @@ public Vocabulary getVocabulary() {
-        case "<=": return "le";
-
-        case ">>": return "double-gt";
-        case "<<": return "double-lt";
-        case ">>>": return "triple-gt";
-        case "<<<": return "triple-lt";
-
-        case "=": return "eq";
-        case "==": return "double-eq";
-        case "===": return "triple-eq";
-        case "!=": return "not-eq";
-
-        case "&": return "amp";
-        case "&&": return "double-amp";
-        case "|": return "pipe";
-        case "||": return "double-pipe";
-
-        case "*": return "star";
-        case "**": return "double-star";
-
-        case "+": return "plus";
-        case "++": return "double-plus";
-        case "-": return "minus";
-        case "--": return "double-minus";
-
-        case "->": return "rarrow";
-        case "<-": return "larrow";
-
-        default:
-            return null;
-        }
+        Map<String, String> mapping = new HashMap<>();
+        mapping.put("!", "bang");
+        mapping.put("!!", "double-bang");
+        mapping.put("?", "question");
+        mapping.put("??", "double-question");
+        mapping.put("?:", "elvis");
+        mapping.put("?.", "question-dot");
+        mapping.put(":", "colon");
+        mapping.put(";", "semi");
+        mapping.put(",", "comma");
+        mapping.put("(", "lparen");
+        mapping.put(")", "rparen");
+        mapping.put("[", "lbracket");
+        mapping.put("]", "rbracket");
+        mapping.put("{", "lbrace");
+        mapping.put("}", "rbrace");
+        mapping.put("_", "underscore");
+        mapping.put(".", "dot");
+        mapping.put("..", "double-dot");
+        mapping.put("...", "ellipsis");
+        mapping.put("@", "at-symbol");
+        mapping.put("$", "dollar");
+        mapping.put("\\", "backslash");
+        mapping.put("/", "slash");
+        mapping.put("//", "double-slash");
+        mapping.put("`", "backtick");
+        mapping.put("'", "squote");
+        mapping.put("\"", "dquote");
+        mapping.put("\"\"\"", "triple-quote");
+        mapping.put(">", "gt");
+        mapping.put(">=", "ge");
+        mapping.put("<", "lt");
+        mapping.put("<=", "le");
+        mapping.put(">>", "double-gt");
+        mapping.put("<<", "double-lt");
+        mapping.put(">>>", "triple-gt");
+        mapping.put("<<<", "triple-lt");
+        mapping.put("=", "eq");
+        mapping.put("==", "double-eq");
+        mapping.put("===", "triple-eq");
+        mapping.put("!=", "not-eq");
+        mapping.put("&", "amp");
+        mapping.put("&&", "double-amp");
+        mapping.put("|", "pipe");
+        mapping.put("||", "double-pipe");
+        mapping.put("*", "star");
+        mapping.put("**", "double-star");
+        mapping.put("+", "plus");
+        mapping.put("++", "double-plus");
+        mapping.put("-", "minus");
+        mapping.put("--", "double-minus");
+        mapping.put("->", "rarrow");
+        mapping.put("<-", "larrow");
+
+        return mapping.getOrDefault(name, null);
     }
```

Comment:
It's great that using Map instead of switch can improve the readability and maintainability of your code, especially when you need to make a lot of branching judgments. Using Map can make the code more concise and easy to understand. It is easy to maintain and extend.